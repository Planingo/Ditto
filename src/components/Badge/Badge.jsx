import React from 'react';
import { Badge as AntdBadge } from 'antd'

/**
 * 
 * Documentations andt du composant : https://ant.design/components/badge/
 * 
 */
export default function Badge({
  ...props
}){
  return <AntdBadge {...props} />
}
