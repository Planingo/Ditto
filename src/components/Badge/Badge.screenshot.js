import { ReactScreenshotTest } from 'react-screenshot-test'
import React from 'react';
import Badge from './Badge';
import 'antd/dist/antd.css';

describe("Screenshot", () => {
  const badge = (
    <Badge color="#f50" text="#f50" />
  );

    ReactScreenshotTest
    .create("Badge")
    .viewport("Desktop", {
      width: 500,
      height: 100
    })
    .shoot("Badge", badge)
    .run();
})
