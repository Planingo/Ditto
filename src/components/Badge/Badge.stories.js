import React from 'react';
import results from '../../../.jest-test-results.json';
import { withTests } from '@storybook/addon-jest';
import { Badge } from './Badge';

export default {
  component: Badge,
  title: 'Atomes/Badge',
  decorators: [withTests({ results })],
};

const Template = args => <Badge {...args} />;

export const Default = Template.bind({});

Default.args = {
    count: 12
}

Default.argTypes = {
    color: {
        control: {
            type: 'color'
        },
    },
    count: {
        control: {
            type: 'number'
        }
    },
    dot: {
        control: {
            type: 'boolean'
        }
    },
    offset: {
        control: {
            type: "object",
        }
    },  
    overflowCount: {
        control: {
            type: 'number'
        }
    },
    showZero: {
        control: {
            type: 'boolean'
        }
    },
    size: {
        control: {
            type: 'select',
            options: [
                'default',
                'small'
            ]
        }
    },
    status: {
        control: {
            type: 'select',
            options: [
                'success',
                'processing',
                'default',
                'error',
                'warning'
            ]
        }
    },
    text: {
        control: {
            type: 'text',
        }
    },
    title: {
        control: {
            type: 'text'
        }
    }
};
