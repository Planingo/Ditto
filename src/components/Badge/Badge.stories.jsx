import React from 'react';
import { Badge } from './Badge';

export default {
  component: Badge,
  title: 'Atomes/Badge',
};

const Template = args => <Badge {...args} />;

export const Default = Template.bind({});

Default.args = {
    count: 12
}

Default.argTypes = {
    color: {
        control: {
            type: 'color'
        },
    },
    count: {
        control: {
            type: 'number'
        }
    },
    dot: {
        control: {
            type: 'boolean'
        }
    },
    offset: {
        control: {
            type: "object",
        }
    },  
    overflowCount: {
        control: {
            type: 'number'
        }
    },
    showZero: {
        control: {
            type: 'boolean'
        }
    },
    size: {
        control: {
            type: 'select',
            options: [
                'default',
                'small'
            ]
        }
    },
    status: {
        control: {
            type: 'select',
            options: [
                'success',
                'processing',
                'default',
                'error',
                'warning'
            ]
        }
    },
    text: {
        control: {
            type: 'text',
        }
    },
    title: {
        control: {
            type: 'text'
        }
    }
};
