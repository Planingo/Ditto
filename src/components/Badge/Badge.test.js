import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import { Badge } from './Badge';

describe('Badge component', () => {
  const badge = (
    <Badge color="#f50" text="#f50" onClick={console.log}/>
  );

  it('Badge content text', () => {
    const { baseElement } = render(badge);
    expect(baseElement).toHaveTextContent("#f50");
  });

});