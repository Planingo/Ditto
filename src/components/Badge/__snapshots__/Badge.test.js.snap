// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component Snapshot badge 1`] = `
<body>
  <div>
    <span
      class="ant-badge ant-badge-status ant-badge-not-a-wrapper css-dev-only-do-not-override-j0nf2s"
    >
      <span
        class="ant-badge-status-dot ant-badge-status-default"
        style="color: rgb(255, 85, 0); background: rgb(255, 85, 0);"
      />
      <span
        class="ant-badge-status-text"
      >
        #f50
      </span>
    </span>
  </div>
</body>
`;
